solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -976,9 +975,6 @@ private void loadClusterProperties() {
           this.clusterProperties = ClusterProperties.convertCollectionDefaultsToNestedFormat(properties);
           log.debug("Loaded cluster properties: {}", this.clusterProperties);
 
-          // Make the urlScheme globally accessible
-          UrlScheme.INSTANCE.setUrlScheme(getClusterProperty(ZkStateReader.URL_SCHEME, HTTP));
```

Comment:
We don't want to set the Singleton `urlScheme` in the `ZkStateReader` since that runs on the client side too, which would force all clients to use the same scheme if multiple are talking to different clusters (or it might flip-flop around too, which would be really bad). It's best for the "client" side to get back whatever is stored from the server, which is why we have to store `base_url` in the replica state.

Reply:
This has now been found in the wild, see: https://issues.apache.org/jira/browse/SOLR-15364